packages/csharp/ArmoniK.Api.Client/EventsClientExt.cs

```diff
@@ -80,75 +81,85 @@ private static FiltersAnd ResultsFilter(string resultId)
-                {
-                  break;
-                }
-              }
+                                                       break;
+                                                     }
+                                                   }
 
-              if (resp.ResultStatusUpdate.Status == ResultStatus.Aborted)
-              {
-                throw new ResultAbortedException($"Result {resp.ResultStatusUpdate.ResultId} has been aborted");
-              }
-            }
+                                                   if (resp.ResultStatusUpdate.Status == ResultStatus.Aborted)
+                                                   {
+                                                     throw new ResultAbortedException($"Result {resp.ResultStatusUpdate.ResultId} has been aborted");
+                                                   }
+                                                 }
 
-            if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.NewResult && resultsNotFound.Contains(resp.NewResult.ResultId))
-            {
-              if (resp.NewResult.Status == ResultStatus.Completed)
-              {
-                resultsNotFound.Remove(resp.NewResult.ResultId);
-                if (!resultsNotFound.Any())
-                {
-                  break;
-                }
-              }
+                                                 if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.NewResult &&
+                                                     resultsNotFound.Contains(resp.NewResult.ResultId))
+                                                 {
+                                                   if (resp.NewResult.Status == ResultStatus.Completed)
+                                                   {
+                                                     resultsNotFound.Remove(resp.NewResult.ResultId);
+                                                     if (!resultsNotFound.Any())
+                                                     {
+                                                       break;
+                                                     }
+                                                   }
 
-              if (resp.NewResult.Status == ResultStatus.Aborted)
-              {
-                throw new ResultAbortedException($"Result {resp.NewResult.ResultId} has been aborted");
-              }
-            }
-          }
-        }
-        catch (OperationCanceledException)
-        {
-        }
-        catch (RpcException)
-        {
-        }
-      }
-    }
+                                                   if (resp.NewResult.Status == ResultStatus.Aborted)
+                                                   {
+                                                     throw new ResultAbortedException($"Result {resp.NewResult.ResultId} has been aborted");
+                                                   }
+                                                 }
```

Comment:
You can handle both cases without repetition by extracting the "status" and "resultId" first depending on the event type, then do the checks and throw if necessary